From 978cc08a393b7d5d0043bf7f4d33f0e33b2b18d8 Mon Sep 17 00:00:00 2001
From: Mikko Rapeli <mikko.rapeli@linaro.org>
Date: Thu, 1 Aug 2024 13:58:36 +0000
Subject: [PATCH 1/3] compile.mk: use CFLAGS from environment

Users can set CFLAGS just like AFLAGS, CC,
LD etc and expect them to be used. It's ok to amend
to them but overwriting should not be done.
Build environment like yocto expect that these
variables are used to call the compiler etc tools.
Linux distro build environments usually set
these variables.

Helps to remove build time paths from generated binaries
since mappings to remove them can be set by the distro
build system in CFLAGS automatically for each SW component
in the build.

Reviewed-by: Jerome Forissier <jerome.forissier@linaro.org>
Signed-off-by: Mikko Rapeli <mikko.rapeli@linaro.org>
---
 mk/compile.mk | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

Upstream-Status: Backport

diff --git a/mk/compile.mk b/mk/compile.mk
index b3d807ba4..0de7ea259 100644
--- a/mk/compile.mk
+++ b/mk/compile.mk
@@ -80,7 +80,8 @@ comp-compiler-$2 := $$(CC$(sm))
 comp-flags-$2 = $$(filter-out $$(CFLAGS_REMOVE) $$(cflags-remove) \
 			      $$(cflags-remove-$$(comp-sm-$2)) \
 			      $$(cflags-remove-$2), \
-		   $$(CFLAGS$$(arch-bits-$$(comp-sm-$2))) $$(CFLAGS_WARNS) \
+		   $$(CFLAGS$$(arch-bits-$$(comp-sm-$2))) $$(CFLAGS) \
+		   $$(CFLAGS_WARNS) \
 		   $$(comp-cflags$$(comp-sm-$2)) $$(cflags$$(comp-sm-$2)) \
 		   $$(cflags-lib$$(comp-lib-$2)) $$(cflags-$2))
 ifeq ($C,1)
-- 
2.34.1

